pkg/dns/source/reconciler.go

```diff
@@ -363,6 +364,24 @@ func (this *sourceReconciler) Delete(logger logger.LogContext, obj resources.Obj
 		return reconcile.Delay(logger, nil)
 	}
 
+	// wait for deletion of all dns entries
+	start := time.Now()
+	for _, s := range slaves {
+		for {
+			_, err := s.GetResource().Get(s.Data())
+			if err != nil {
+				if errors.IsNotFound(err) {
+					break
+				}
+				return reconcile.Delay(logger, fmt.Errorf("get dns entry failed %s: %s", s.ObjectName(), err))
```

Comment:
Tiny detail to wrap the error:
```suggestion
				return reconcile.Delay(logger, fmt.Errorf("get dns entry failed %s: %w", s.ObjectName(), err))
```